### PR TITLE
feat: highlight conversation search results matches (AR-2575)

### PR DESCRIPTION
**Pull request by trOnk12**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2575" title="AR-2575" target="_blank">StoryAR-2575</a>  search conversation: highlight matched text
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Highlight the matches when the user searches for a  specific conversation

image

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1146/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4895556)(4895556)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4895556)(4895556)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1146/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4895712)(4895712)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4895712)(4895712)-dev-release.apk)

**Review by alexandreferris — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt [hunk 1]
```diff
@@ -106,14 +106,15 @@ class ConversationListViewModel @Inject constructor(
...
-                            emit(searchConversation(conversationDetails, searchQuery))
+                            emit(searchConversation(conversationDetails, searchQuery) to searchQuery)
                         }
                     }
-                }.collect { conversationSearchResult ->
+                }.collect { (conversationSearchResult, searchQuery) ->
                     conversationListState = conversationListState.copy(
-                        conversationSearchResult = conversationSearchResult.toConversationsFoldersMap().toImmutableMap()
+                        conversationSearchResult = conversationSearchResult.toConversationsFoldersMap().toImmutableMap(),
+                        searchQuery = searchQuery
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:119]: Maybe extract all this code to a separated function?

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/call/CallsScreen.kt [hunk 2]
```diff
@@ -66,7 +65,8 @@ fun CallContent(
                 openMenu = onEditConversationItem,
                 openUserProfile = onOpenUserProfile,
                 openNotificationsOptions = openConversationNotificationsSettings,
-                joinCall = onJoinCall
+                joinCall = onJoinCall,
+                searchQuery = "test"
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/call/CallsScreen.kt]: Leftover from manual testing?

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/call/CallsScreen.kt [hunk 3]
```diff
@@ -80,7 +80,8 @@ fun CallContent(
...
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/call/CallsScreen.kt]: Leftover?

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/common/ConversationTitle.kt [hunk 4]
```diff
@@ -12,26 +12,32 @@ import androidx.compose.ui.text.style.TextOverflow
...
-        )
+        if(searchQuery.isEmpty()) {
+            Text(
+                text = name,
+                style = MaterialTheme.wireTypography.body02,
+                overflow = TextOverflow.Ellipsis,
+                maxLines = 1,
+                modifier = Modifier.weight(weight = 1f, fill = false)
+            )
+        }else{
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/common/ConversationTitle.kt]: Can we apply formatting & spacing?

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/mention/MentionScreen.kt [hunk 5]
```diff
@@ -78,7 +79,8 @@ private fun MentionContent(
                 openMenu = onEditConversationItem,
                 openUserProfile = onOpenUserProfile,
                 openNotificationsOptions = openConversationNotificationsSettings,
-                joinCall = onJoinCall
+                joinCall = onJoinCall,
+                searchQuery = "test"
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/mention/MentionScreen.kt]: leftover

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/mention/MentionScreen.kt [hunk 6]
```diff
@@ -64,7 +64,8 @@ private fun MentionContent(
...
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/mention/MentionScreen.kt]: leftover

**Review by gongracr — APPROVED** [review]
Looking in good shape 💪  Just minor code formatting bits.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1146/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1146/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1146/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4921331)(4921331)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4921331)(4921331)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1146/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4945098)(4945098)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4945098)(4945098)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1146/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4945199)(4945199)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4945199)(4945199)-dev-release.apk)

**Pull request merged**